### PR TITLE
sharding tutorial for torchrec

### PR DESCRIPTION
**Pull request by YazhiGao**
This commit add the tutorial for exploring torchrec sharding for
distributed model parallelism for embedding tables in large scale deep
learning recommendation systems. We exploring the following sharding:

- table-wise
- row-wise
- column-wise
- data-parallel

### CODE REVIEW

**Comment by YazhiGao** [comment]
@colin2328 @brianjo

**Review by colin2328 — APPROVED** [review]
LGTM, thanks @YazhiGao !

**Review by svekars — APPROVED** [review]

File: advanced_source/sharding.rst [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+The following steps are needed for the Colab runtime to detect the added
+shared libraries. The runtime searches for shared libraries in /usr/lib,
+so we copy over the libraries which were installed in /usr/local/lib/.
+**This is a very necessary step, only in the colab runtime**.
+
+.. code:: python
+
+    !sudo cp /usr/local/lib/lib* /usr/lib/
+
+\*\*Restart your runtime at this point for the newly installed packages
```
Comment (svekars) [inline advanced_source/sharding.rst:47]:
```suggestion
**Restart your runtime at this point for the newly installed packages
```

File: advanced_source/sharding.rst [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+to be seen.\ ** Run the step below immediately after restarting so that
```
Comment (svekars) [inline advanced_source/sharding.rst:48]:
```suggestion
to be seen.** Run the step below immediately after restarting so that
```

**Review by svekars — COMMENTED** [review]
A very minor tweak.

**Review by colin2328 — APPROVED** [review]
thanks

**Pull request merged**